### PR TITLE
[ut] add ut for pkg/util/clusterrole, rbac, sercret,  secret, serviceaccount

### PR DESCRIPTION
**Pull request by ikaven1024**
Signed-off-by: yingjinhui <yingjinhui@didiglobal.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
add ut for  pkg/util/clusterrole, rbac, sercret,  secret, serviceaccount

**Which issue(s) this PR fixes**:
Part #2579

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/util/rbac_test.go [hunk 1]
```diff
@@ -0,0 +1,622 @@
...
+	}{
+		{
+			name: "empty",
+			args: args{
+				allSubjects: nil,
+			},
+			want: nil,
+		},
+		{
+			name: "empty",
```
Comment (XiShanYongYe-Chang) [inline pkg/util/rbac_test.go:343]: This case name seems not correct.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks! Other parts look good to me.

File: pkg/util/secret.go [hunk 2]
```diff
@@ -8,40 +8,11 @@ import (
...
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	kubeclient "k8s.io/client-go/kubernetes"
 	"k8s.io/klog/v2"
 )
 
-// GetTargetSecret will get secrets(type=targetType, namespace=targetNamespace) from a list of secret references.
-func GetTargetSecret(client kubeclient.Interface, secretReferences []corev1.ObjectReference, targetType corev1.SecretType, targetNamespace string) (*corev1.Secret, error) {
```
Comment (RainbowMango) [inline pkg/util/secret.go:18]: Nice, we should always cleanup unused functions.

File: pkg/util/secret_test.go [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+				secret: makeSecret("test"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "update success",
+			args: args{
+				client: fake.NewSimpleClientset(makeSecret("test")),
+				secret: makeSecret("test"),
+			},
```
Comment (RainbowMango) [inline pkg/util/secret_test.go:40]: I think there is a problem with the `CreateOrUpdate` function, as I mentioned on https://github.com/karmada-io/karmada/pull/2596#discussion_r1006333298.
I'm surprised the test can't find it.

**Review by RainbowMango — COMMENTED** [review]
Nice work!
Just a question need to confirm.

File: pkg/util/secret_test.go (hunk 3 above)
Comment (ikaven1024) [reply pkg/util/secret_test.go:40]: fake client doesn't check the resourceVersion.

https://github.com/karmada-io/karmada/blob/531d6bf2acbe9dd572ef5f09437ad713bc2f41e1/vendor/k8s.io/client-go/testing/fixture.go#L116-L126

Comment (RainbowMango) [reply pkg/util/secret_test.go:40]: OK.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**